src/java_bytecode/java_types.h

```diff
@@ -313,6 +319,64 @@ inline const typet &java_generic_class_type_bound(size_t index, const typet &t)
   return gen_type.subtype();
 }
 
+/// Type to hold a Java class that is implicitly generic, e.g., an inner
+/// class of a generic outer class or a derived class of a generic base
+/// class. Extends the java class type.
+class java_implicitly_generic_class_typet:public java_class_typet
+{
+public:
+  typedef std::vector<java_generic_parametert> implicit_generic_typest;
+
+  explicit java_implicitly_generic_class_typet(
+    const java_class_typet &class_type,
+    const implicit_generic_typest &generic_types)
+    : java_class_typet(class_type)
```

Comment:
Why does this need to be called? Are you trying to achieve something in particular? Normally you wouldn't want to pass a base class object to the derived class constructor, and call the superclass' constructor like this, to my understanding. If you leave it be, without calling it, the superclass' default constructor should be called automatically, which I'm assuming should do some sensible initialisation.

Another problem this can create is that you can pass a subclass of `java_class_typet` to this constructor and it will essentially clobber internal state that doesn't match the `java_class_typet` which can cause some very nasty bugs in the future.

Reply:
After further discussion, this will stay as it is.